express/parsers/apps/espresso/parser.py

```diff
@@ -22,13 +24,18 @@ class EspressoParser(BaseParser, IonicDataMixin, ElectronicDataMixin, Reciprocal
     """
 
     def __init__(self, *args, **kwargs):
-        super(EspressoParser, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
         self.work_dir = self.kwargs["work_dir"]
         self.stdout_file = self.kwargs["stdout_file"]
         self.txt_parser = EspressoTXTParser(self.work_dir)
+
+        # use packaging parser for accurate X.x.x version comparison.
         self.xml_parser = EspressoXMLParser(self.find_xml_file())
+        if self.version:
+            if version.parse(self.version) > version.parse("6.4"):
+                self.xml_parser = EspressoXMLParserV7(self.find_xml_file(settings.XML_DATA_FILE_NEW))
```

Comment:
The "factory" pattern could work here - I'd create `getXMLParser` function and then switch based on the version